include/cudaq/Optimizer/CodeGen/Passes.h

```diff
@@ -27,16 +27,21 @@ namespace cudaq::opt {
 std::unique_ptr<mlir::Pass> createConvertToQIRPass();
 void registerConvertToQIRPass();
 
-/// Convert (generic) QIR to the Base Profile QIR for a specific target.
-/// TODO: Decide how to convey the selected target information.
-void addBaseProfilePipeline(mlir::OpPassManager &pm);
-void registerBaseProfilePipeline();
-
-// Use the addBaseProfilePipeline() for the following passes.
-std::unique_ptr<mlir::Pass> createQIRToBaseProfilePass();
-std::unique_ptr<mlir::Pass> verifyBaseProfilePass();
-std::unique_ptr<mlir::Pass> createBaseProfilePreparationPass();
-std::unique_ptr<mlir::Pass> createConvertToQIRFuncPass();
+/// Convert (generic) QIR to the profile-specific QIR for a specific target.
+/// @param pm Pass Manager to add QIR passes to
+/// @param convertTo Expected to be "qir-base" or "qir-adaptive" (comes from the
+/// cudaq-translate command line `--convert-to` parameter)
+void addQIRProfilePipeline(mlir::OpPassManager &pm,
+                           const std::string &convertTo);
```

Comment:
Instead of `const std::string &` we should probably use `llvm::StringRef` to be consistent.

Reply:
Updated in 09a91f29. Let me know if this is what you had in mind.

Reply:
This looks good. Thanks for making these changes.